test/performance/range/views/view_drop_benchmark.cpp

```diff
@@ -62,31 +62,66 @@ void sequential_read(benchmark::State & state)
 
 BENCHMARK_TEMPLATE(sequential_read, std::string, void);
 BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(std::views::drop));
+#ifdef SEQAN3_DEPRECATED_310
```

Comment:
I wonder if it had been easier to have another `seqan3::detail::drop` instance that is not deprecated and to use it in the tests.

Reply:
we want to remove our implementation, so this should be the "right way" to do this. remove the benchmark parts. we could argue to remove the whole benchmark.

we could maybe add a local type-reduced drop adaptor just for this benchmark, like this

```cpp
inline constexpr type_reduced_drop = seqan3::views::type_reduce | std::views::drop;
```

Reply:
Mhm, I meant something like:

```cpp
namespace seqan3::detail
{

inline constexpr auto drop = detail::drop_fn{};

}
```

Then you have `seqan3::detail::drop` which is not deprecated, and `seqan3::views::drop` which is deprecated.
Now you could use `seqan3::detail::drop` in tests and benchmarks. When we finally remove it, we can remove all `seqan3::detail::drop` occurrences.
Seems easier than doing `pragma` everywhere.

Reply:
this does not work; you can only pipe range_adaptor_closures

Reply:
Why not?
`detail::drop` is the same as `views::drop`.
So you can just use `detail::drop` in tests, instead of `pragma`ing them

Reply:
This would work:

```patch
diff --git a/test/performance/range/views/view_drop_benchmark.cpp b/test/performance/range/views/view_drop_benchmark.cpp
index a3e20bcc9..9ad8356aa 100644
--- a/test/performance/range/views/view_drop_benchmark.cpp
+++ b/test/performance/range/views/view_drop_benchmark.cpp
@@ -16,6 +16,7 @@
 
 #include <seqan3/range/views/drop.hpp>
 #include <seqan3/range/views/single_pass_input.hpp>
+#include <seqan3/range/views/type_reduce.hpp>
 
 // ============================================================================
 //  sequential_read
@@ -60,68 +61,41 @@ void sequential_read(benchmark::State & state)
     [[maybe_unused]] volatile uint8_t dummy2 = dummy;
 }
 
+struct type_reduced_drop_fn
+{
+    constexpr auto operator()(int drop_size) const
+    {
+        return seqan3::views::type_reduce | std::views::drop(drop_size);
+    }
+};
+
 BENCHMARK_TEMPLATE(sequential_read, std::string, void);
 BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(std::views::drop));
-#ifdef SEQAN3_DEPRECATED_310
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(seqan3::views::drop));
-#pragma GCC diagnostic pop
-#endif // SEQAN3_DEPRECATED_310
+BENCHMARK_TEMPLATE(sequential_read, std::string, type_reduced_drop_fn);
 
 BENCHMARK_TEMPLATE(sequential_read, std::vector<uint8_t>, void);
 BENCHMARK_TEMPLATE(sequential_read, std::vector<uint8_t>, decltype(std::views::drop));
```

Reply:
Yes, but I meant:

```cpp
diff --git a/include/seqan3/range/views/drop.hpp b/include/seqan3/range/views/drop.hpp
index d1680ca9..5fd8c42e 100644
--- a/include/seqan3/range/views/drop.hpp
+++ b/include/seqan3/range/views/drop.hpp
@@ -99,6 +99,9 @@ struct drop_fn
     }
 };
 
+//!brief Instance for usage until removed.
+inline constexpr auto drop = drop_fn{};
+
 } // namespace seqan3::detail
 
 // ============================================================================
diff --git a/test/performance/range/views/view_drop_benchmark.cpp b/test/performance/range/views/view_drop_benchmark.cpp
index fe645ef4..6545fec8 100644
--- a/test/performance/range/views/view_drop_benchmark.cpp
+++ b/test/performance/range/views/view_drop_benchmark.cpp
@@ -62,7 +62,7 @@ void sequential_read(benchmark::State & state)
 
 BENCHMARK_TEMPLATE(sequential_read, std::string, void);
 BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(std::views::drop));
-BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(seqan3::views::drop));
+BENCHMARK_TEMPLATE(sequential_read, std::string, decltype(seqan3::detail::drop));
 
 BENCHMARK_TEMPLATE(sequential_read, std::vector<uint8_t>, void);
 BENCHMARK_TEMPLATE(sequential_read, std::vector<uint8_t>, decltype(std::views::drop));

```
(and replacing `seqan3::views::drop` with `seqan3::detail::drop` in the unit test)

And once we remove `seqan3::views::drop`, we can remove the test as a whole and the benchmarks with `seqan3::detail::drop`.

Reply:
Ah okay :)

I think I want to keep those benchmarks. Maybe adding type_reduce later would also be cool.